docs/pallas/grid_blockspec.md

```diff
@@ -233,3 +236,46 @@ See [the Pallas TPU documentation](https://jax.readthedocs.io/en/latest/pallas/t
  [309 309 309 319 319 319]]
 
 ```
+
+A `None` value appearing as a dimension value in the `block_shape` behaves
+as the value `1`, except that the corresponding
+block axis is squeezed. In the example below, observe that the
+shape of the `o_ref` is (2,) when the block shape was specified as
+`(None, 2)` (the leading dimension was squeezed).
+
+```python
+>>> def kernel(o_ref):
+...   assert o_ref.shape == (2,)
+...   o_ref[...] = jnp.full((2,), 10 * pl.program_id(1) + pl.program_id(0))
+>>> pl.pallas_call(kernel,
+...                jax.ShapeDtypeStruct((3, 4), dtype=np.int32),
+...                out_specs=pl.BlockSpec((None, 2), lambda i, j: (i, j)),
+...                grid=(3, 2), interpret=True)()
+Array([[ 0,  0, 10, 10],
+       [ 1,  1, 11, 11],
+       [ 2,  2, 12, 12]], dtype=int32)
+
+```
+
+When we construct a `BlockSpec` we can use the value `None` for the
+`block_shape` parameter, in which case the shape of the overall array
+is used. And if we use the value `None` for the `index_map` parameter
+then a default index map function that returns a tuple of zeros is
+used: `index_map=lambda *invocation_indices: (0,) * len(block_shape)`.
```

Comment:
Oh wow I had no idea about this. I don't think this is even desirable? It makes no sense to specify a BlockSpec without an index function. Can we not document this behavior and perhaps restrict it?

Reply:
I agree that there is too much optionality in the API. We should be allowed to skip BlockSpec, but if we have them they should include the block_shape and index_map. If there is agreement about that, I would do it in a different PR, because it changes the API.

Reply:
@sharadmv Agree with less optionality?